### ISSUE TITLE
Installing forum_ruby_version BUILD FAILED...

### ISSUE THREAD

**Issue by ClaudeCoulombe** [openedx-unsupported/configuration#459]
I've already successfully installed previous version of edX on an AWS instance based on Linux Ubuntu 12.04 (ami-69f5a900) following the procedure 
edX Ubuntu 12.04 Installation (https://github.com/edx/configuration/wiki/edX-Ubuntu-12.04-Installation).

Trying to install the new edX version, I've got build fatal problem that seems related to the Ruby versions used by the forum tool which is different from version of Ruby used for other modules.

TASK: [rbenv | check ruby {{forum_ruby_version}} installed] ******************\* 
failed: [localhost] => {"changed": true, "cmd": "rbenv versions | grep 1.9.3-p448 ", "delta": "0:00:00.105147", "end": "2013-11-15 04:40:30.269642", "item": "", "rc": 1, "start": "2013-11-15 04:40:30.164495"}
...ignoring

TASK: [rbenv | install ruby {{forum_ruby_version}}] **************************\* 
failed: [localhost] => {"changed": true, "cmd": "rbenv install 1.9.3-p448 ", "delta": "0:07:58.251972", "end": "2013-11-15 04:48:28.987009", "item": "", "rc": 1, "start": "2013-11-15 04:40:30.735037"}
stderr: Downloading yaml-0.1.4.tar.gz...
-> http://dqw8nmjcqpjn7.cloudfront.net/36c852831d02cf90508c29852361d01b
Installing yaml-0.1.4...
Installed yaml-0.1.4 to /edx/app/forum/.rbenv/versions/1.9.3-p448

Downloading ruby-1.9.3-p448.tar.gz...
-> http://dqw8nmjcqpjn7.cloudfront.net/a893cff26bcf351b8975ebf2a63b1023
Installing ruby-1.9.3-p448...

BUILD FAILED

Inspect or clean up the working tree at /tmp/ruby-build.20131115044031.4113
Results logged to /tmp/ruby-build.20131115044031.4113.log

Last 10 log lines:
    from ./tool/rbinstall.rb:238:in `each'
    from ./tool/rbinstall.rb:238:in`install_recursive'
    from ./tool/rbinstall.rb:384:in `block in <main>'
    from ./tool/rbinstall.rb:597:in`call'
    from ./tool/rbinstall.rb:597:in `block (2 levels) in <main>'
    from ./tool/rbinstall.rb:594:in`each'
    from ./tool/rbinstall.rb:594:in `block in <main>'
    from ./tool/rbinstall.rb:590:in`each'
    from ./tool/rbinstall.rb:590:in `<main>'
make: **\* [do-install-all] Error 1

FATAL: all hosts have already failed -- aborting

additional remarks:

I've started from scratch. 

The Studio seems running fine but not the LMS since it depends on the forum tool, I presumed.

The same problem occurs when I started using ami-d0f89fb9 which was suggested by John Jarvis

ami-d0f89fb9

**Comment by ClaudeCoulombe** [comment]
Greetings,

Following the above open issue...

I've installed ruby-1.9.3-p448 using rvm on an Amazon Ubuntu 12.04 Ami

rvm install 1.9.3-p448

So the version was installed as the next ls command shows:

ls /home/ubuntu/.rvm/wrappers/
default  ruby-1.9.3-p448  ruby-1.9.3-p448@global  ruby-2.0.0-p353  ruby-2.0.0-p353@global

But I've still a problem with forum_ruby_version as shown below

TASK: [rbenv | install ruby {{forum_ruby_version}}] **************************\* 
failed: [localhost] => {"changed": true, "cmd": "rbenv install 1.9.3-p448 ", "delta": "0:07:13.131717", "end": "2013-11-25 18:56:32.827188", "item": "", "rc": 1, "start": "2013-11-25 18:49:19.695471"}
stderr: Downloading yaml-0.1.4.tar.gz...
-> http://dqw8nmjcqpjn7.cloudfront.net/36c852831d02cf90508c29852361d01b
Installing yaml-0.1.4...
Installed yaml-0.1.4 to /edx/app/forum/.rbenv/versions/1.9.3-p448

Downloading ruby-1.9.3-p448.tar.gz...
-> http://dqw8nmjcqpjn7.cloudfront.net/a893cff26bcf351b8975ebf2a63b1023
Installing ruby-1.9.3-p448...

BUILD FAILED

Inspect or clean up the working tree at /tmp/ruby-build.20131125184919.26581
Results logged to /tmp/ruby-build.20131125184919.26581.log

Last 10 log lines:
Enclosing class/module 'mDigest' for class SHA1 not known
Enclosing class/module "cASN1ObjectId" for alias short_name sn not known
Enclosing class/module "cASN1ObjectId" for alias long_name ln not known

Generating RI format into /tmp/ruby-build.20131125184919.26581/ruby-1.9.3-p448/.ext/rdoc...
uh-oh! RDoc had a problem:
No space left on device - ./RDoc/Error

run with --debug for full backtrace
make: **\* [rdoc] Error 1

FATAL: all hosts have already failed -- aborting

---

Does someone have a workaround?

Thanks

Claude

**Comment by ClaudeCoulombe** [comment]
So I've revisited the problem...

Now, I've used rbenv to manage the different versions of Ruby.
It seems rbenv was the tool used by people at edX and there are well known conflicts between rvm and rbenv.

sudo apt-get install rbenv

git clone https://github.com/sstephenson/ruby-build.git ~/.rbenv/plugins/ruby-build

git clone https://github.com/sstephenson/ruby-build.git

cd ruby-build

sudo ./install.sh

rbenv init -

rbenv install 1.9.3-p374

rbenv install 1.9.3-p448

rbenv install 2.0.0-p353

rbenv global 1.9.3-p448

So, I've no more problem with Ruby.
